dynamics/drag_dynamics.py

```diff
@@ -52,3 +55,70 @@ def drag_jacobian(x: np.ndarray, drag_const: float, latest_epoch: Epoch) -> np.n
     da_drag_dv = density * drag_const * ((np.eye(3) * v_norm) - np.outer(x[3:6], x[3:6]) / v_norm)
 
     return da_drag_dv
+
+
+def drag_scalar_estimate(x: np.ndarray, drag_const: float) -> np.ndarray:
```

Comment:
Can you change all the inputs here to be specifically what the functions require, not just the state vector? These functions are low level building blocks and should not have any knowledge of higher level concepts like the state vector.

I can't even tell what `x[9]` is in this function without having to dig through the EKF code, which I think illustrates why separation of concerns like this is essential. Same thing goes for the functions that follow in this file.